### DISCUSSION TITLE
Custom Hook between collection !?

### DISCUSSION THREAD

**Discussion by hugopvl** [directus/directus#10087] [Archive]
Hey guys,

I'm here with a situation, to save colleagues work I would like to know if it is possible to do the following:

I have a collection with several fields,

collection 1:
image, field 1, field 2, field 3, field 4

so when the user uploads the photo to the image field, from collocation 1,

would like to create a new item in another collection, collection 2,

with the image, field 1 and field 2 who comes from collection 1.

it is possible?
what is the best way to do it?

Has anyone ever made a "costum hook" with something like that?

thank you guys,

**Comment by keesvanbemmel** [comment]
This example is far from neat but it should give you an idea:
```
let hookContext;

async function copyItemFromOneToTwo(input, context) {
  if (input.payload.somefile) {
    const { services } = hookContext;
    const { schema } = context;
    const collection1Service = new services.ItemsService('collection1', { schema });
    const collection2Service = new services.ItemsService('collection2', { schema });

    const itemToCopy = await collection1Service.readOne(input.keys[0], {
      fields: ['field1', 'field2', 'somefile'],
    });

    await collection2Service.createOne({
      field1: itemToCopy.field1,
      field2: itemToCopy.field2,
      somefile: itemToCopy.somefile,
    });
  }
  return input;
}

module.exports = function registerHook({ action }, incomingHookContext) {
  hookContext = incomingHookContext;

  action('collection1.items.update', copyItemFromOneToTwo);
};
```
